bin/templates/scripts/cordova/lib/prepare.js

```diff
@@ -257,10 +257,11 @@ function handleBuildSettings(platformConfig, locations) {
     return Q();
 }
 
-function handleIcons(projectConfig, platformRoot) {
+function handleIcons(projectConfig, platformRoot, locations) {
 
     var icons = projectConfig.getIcons('ios');
-    var appRoot = projectConfig.path;
+    var appRoot = path.dirname(projectConfig.path);
+    var originalName = path.basename(locations.xcodeCordovaProj);
```

Comment:
no need to create a new variable originalName or add 3rd parameter, use the value already provided here self.locations.xcodeCordovaProj